### PR TITLE
Add logging for bll

### PR DESCRIPTION
**Pull request by darhan1168**

### CODE REVIEW

**Comment by darhan1168** [comment]
[Add logging for BLL](https://trello.com/c/ZcnJJKHy/59-add-logging-for-bll)

File: CourseSystem/BLL/Services/EmailService.cs [hunk 1]
```diff
@@ -175,19 +125,53 @@ public async Task<Result<bool>> SendInvitationToStudents(Dictionary<string, stri
...
+            if (appUser == null)
+            {
+                _logger.LogError("Failed to {action}, {entity} was null!", MethodBase.GetCurrentMethod()?.Name, nameof(appUser));
+
+                return new Result<bool>(false, "Fail to send email");
+            }
+
+            var emailContent = GetEmailSubjectAndBody(emailType, appUser, group, course, file, callBackUrl, tempPassword);
+
+
```
Comment (gavrylenko-daiana) [inline CourseSystem/BLL/Services/EmailService.cs:154]: Extra space

File: CourseSystem/BLL/Services/EmailService.cs [hunk 2]
```diff
@@ -234,31 +223,77 @@ private async Task<Result<bool>> CreateAndSendEmail(List<string> toEmails, strin
...
+                    {
+                        await client.ConnectAsync(_emailSettings.Host, _emailSettings.Port, SecureSocketOptions.SslOnConnect);
+                    }
+                    else if (_emailSettings.UseStartTls)
+                    {
+                        await client.ConnectAsync(_emailSettings.Host, _emailSettings.Port, SecureSocketOptions.StartTls);
+                    }
+
+                    await client.AuthenticateAsync(_emailSettings.UserName, _emailSettings.Password); //ключ доступа от Гугл
+                    await client.SendAsync(emailMessage);
```
Comment (gavrylenko-daiana) [inline CourseSystem/BLL/Services/EmailService.cs:279]: Add logging to the region Email sending

**Review by gavrylenko-daiana — APPROVED** [review]

File: CourseSystem/BLL/Services/EmailService.cs (hunk 1 above)
Comment (darhan1168) [reply CourseSystem/BLL/Services/EmailService.cs]: Fixed

File: CourseSystem/BLL/Services/EmailService.cs (hunk 2 above)
Comment (darhan1168) [reply CourseSystem/BLL/Services/EmailService.cs:278]: Fixed

File: CourseSystem/BLL/Services/AssignmentAnswerService.cs [hunk 3]
```diff
@@ -1,35 +1,46 @@
...
             AppUser appUser, IFormFile? file, DropboxFolders folder)
         {
             if (assignmentAnswer == null)
             {
+                _logger.LogError("Failed to {action}, {entity} was null!", MethodBase.GetCurrentMethod()?.Name, nameof(assignmentAnswer));
+
                 return new Result<bool>(false, "Invalid assignment answer");
             }
 
+            _logger.LogInformation("Forming {action} for user {userId} about assignmentAnswer {assignmentId}",
```
Comment (HerasymovaV) [inline CourseSystem/BLL/Services/AssignmentAnswerService.cs:35]: Strange wording ("Forming assignmentAnswer for user {userId} and/about assignment {assignmentId}")

Comment (darhan1168) [reply CourseSystem/BLL/Services/AssignmentAnswerService.cs]: Fixed

**Review by KatrinaSidorenko — APPROVED** [review]

**Pull request merged**